src/widgets/breadcrumb/breadcrumb.js

```diff
@@ -0,0 +1,194 @@
+}) => (
+  { canRefine, createURL, instantSearchInstance, items, refine },
+  isFirstRendering
+) => {
+  if (isFirstRendering) {
+    renderState.templateProps = prepareTemplateProps({
+      defaultTemplates,
+      templatesConfig: instantSearchInstance.templatesConfig,
+      templates,
+      transformData,
+    });
+    return;
+  }
+
+  const shouldAutoHideContainer = autoHideContainer && !canRefine;
+
+  ReactDOM.render(
+    <Breadcrumb
+      canRefine={canRefine}
+      cssClasses={cssClasses}
+      createURL={createURL}
+      items={items}
+      refine={refine}
+      separator={separator}
+      shouldAutoHideContainer={shouldAutoHideContainer}
+      templateProps={renderState.templateProps}
+    />,
+    containerNode
+  );
+};
+
+const usage = `Usage:
+breadcrumb({
+  container,
+  attributes,
+  [ autoHideContainer=true ],
+  [ cssClasses.{disabledLabel, home, label, root, separator}={} ],
+  [ templates.{home, separator}]
+  [ transformData.{item} ],
+  
+})`;
+
+/**
+ * @typedef {Object} BreadcrumbCSSClasses
+ * @property {string|string[]} [disabledLabel] CSS class to add to the last element of the breadcrumb (which is not clickable).
+ * @property {string|string[]} [home] CSS class to add to the first element of the breadcrumb.
+ * @property {string|string[]} [label] CSS class to add to the text part of each element of the breadcrumb.
+ * @property {string|string[]} [root] CSS class to add to the root element of the widget.
+ * @property {string|string[]} [separator] CSS class to add to the separator.
+ */
+
+/**
+ * @typedef {Object} BreadcrumbTemplates
+ * @property {string|function(object):string} [home='Home'] Label of the breadcrumb's first element.
+ * @property {string|function(object):string} [separator=''] Symbol used to separate the elements of the breadcrumb.
+ */
+
+/**
+ * @typedef {Object} BreadcrumbTransforms
+ * @property {function(object):object} [item] Method to change the object passed to the `item` template
+ */
+
+/**
+ * @typedef {Object} BreadcrumbWidgetOptions
+ * @property {string|HTMLElement} container CSS Selector or HTMLElement to insert the widget.
+ * @property {string[]} attributes Array of attributes to use to generate the breadcrumb.
+ *
+ * You can also use a sort function that behaves like the standard Javascript [compareFunction](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/sort#Syntax).
+ * @property {BreadcrumbTemplates} [templates] Templates to use for the widget.
+ * @property {BreadcrumbTransforms} [transformData] Set of functions to transform the data passed to the templates.
+ * @property {boolean} [autoHideContainer=true] Hides the container when there are no items in the breadcrumb.
+ * @property {BreadcrumbCSSClasses} [cssClasses] CSS classes to add to the wrapping elements.
+ */
+
+/**
+ * The breadcrumb widget is a secondary navigation scheme that allows the user to see where the current page is in relation to the website's hierarchy.
```

Comment:
It might be a bit confusing to talk about website hierarchy because search is only a part of the website. I would go for something like 'facet hierarchy', WDYT?